docs/audit_report/src/side_channels/01_02_ml_dsa.rst

```diff
@@ -0,0 +1,124 @@
+""""""
+ML DSA
+""""""
+
+Analysed variants:
+
+- ML-DSA-44
+- ML-DSA-65
+- ML-DSA-87
+
+For the analysis of ML DSA, a utility was written that calls the functions to be analysed in a similar way to the Botan CLI.
+The following call is used to create the signature:
+
+.. code-block:: cpp
+
+    auto sk = Botan::Dilithium_PrivateKey(rng, mode);
+
+    Botan::PK_Signer sig(sk, rng, "Deterministic");
+    signature = sig.sign_message(message, rng);
+
+
+The Botan library is configured using the following console prompt:
+
+.. code-block::
+
+    ./configure.py --prefix=~/workspace/bsi/DATA/cryptolib/botan/build --cc=gcc \
+    --cc-bin=g++-12 --cc-abi=-fno-plt --disable-modules sm4 --disable-sse2      \
+    --disable-ssse3 --disable-sse4.1 --disable-sse4.2 --disable-avx2            \
+    --disable-bmi2 --disable-rdrand --disable-rdseed --disable-aes-ni           \
+    --disable-sha-ni --disable-altivec --disable-neon --disable-armv8crypto     \
+    --disable-powercrypto --without-os-feature=threads --with-debug-info
+
+The binary is compiled with the `gcc` compiler with the following version:
+
+.. code-block::
+
+    $ g++-12 --version
+    g++-12 (Debian 12.2.0-14) 12.2.0
+
+The host operating system is `Debian GNU/Linux 12 (bookworm)`.
+
+
+**Summary**
+
+No critical leak was identified.
+All leaks found by DATA are unproblematic.
+The reasoning for each identified leak is explained below.
+
+
+**Leak: Hints**
+
+In the analysis with DATA, leaks were detected in the functions `make_hint()` (:srcref:`[src/lib/pubkey/dilithium/dilithium_common]/dilithium_algos.cpp:843|make_hint`),
```

Comment:
The leak in `make_hint` is surprising... Probably compiler optimization shenanigans? Maybe I'll look into this further to see what went wrong. But I agree that hint leaks should be uncritical.